tt_metal/llrt/tt_cluster.cpp

```diff
@@ -247,35 +245,35 @@ void Cluster::get_metal_desc_from_tt_desc(
     }
 }
 
-void Cluster::open_driver(
-    chip_id_t mmio_device_id, const std::set<chip_id_t> &controlled_device_ids, const bool &skip_driver_allocs) {
+void Cluster::open_driver(const bool &skip_driver_allocs) {
     const std::string sdesc_path = get_soc_description_file(this->arch_, this->target_type_);
 
     std::unique_ptr<tt_device> device_driver;
     if (this->target_type_ == TargetDevice::Silicon) {
         // This is the target/desired number of mem channels per arch/device.
         // Silicon driver will attempt to open this many hugepages as channels, and assert if workload uses more than
         // available. Metal currently uses assigns 1 channel per device
-        uint32_t num_host_mem_ch_per_mmio_device = controlled_device_ids.size();
-        if (is_tg_cluster_) {
-            num_host_mem_ch_per_mmio_device = HOST_MEM_CHANNELS;
-        }
+        uint32_t num_host_mem_ch_per_mmio_device = HOST_MEM_CHANNELS;
```

Comment:
just wanted to call out that on N300 this would have previously resolved to 2 (1 host mem channel for the mmio device and 1 for the remote). Now this unconditionally requests the max (4) so each machine would need to have 4 1GB hugepages

Reply:
Yes, I wasn't sure how to rewrite this properly, since it requested for each mmio the number of remote + 1 local chip. 
HOST_MEM_CHANNELS seems to be the same as g_MAX_HOST_MEM_CHANNELS in UMD. Not sure if should have something smarter here, do you have a suggestion? Maybe using min(HOST_MEM_CHANNELS, number_of_chips), which would resolve to a smaller number in cases where cluster has < 4 chips? Or do you think it's fine to request 4 always?

Reply:
Lets go for `min(HOST_MEM_CHANNELS, number_of_chips)` for now otherwise on some machines we might see the warning from umd that prints that X channels requested but only Y available

Reply:
Additional notes:
If we take 2x N300, connected through ethernet, as an example. Previous behavior was that it would create two tt_siliconDevice objects, each requesting 2 num channels.
Now it will be a single object, but for each mmio device it would create 4 channels.
There's a loop inside UMD that does for each mmio device for channel up to m_num_host_mem_channels.

Neither of proposed solution will exactly match previous behavior. However, proposed min(HOST_MEM_CHANNELS, number_of_chips) is slightly closer to previous behavior, so will change to that.